src/main/java/gov/cabinetofice/gapuserservice/web/SpotlightController.java

```diff
@@ -119,4 +119,44 @@ public RedirectView callback(@RequestParam String code, @RequestParam String sta
     }
 
 
+    @GetMapping("/oauth/refresh")
+    public ResponseEntity refresh(final HttpServletRequest httpRequest) throws Exception {
+        log.info("SpotlightController /oauth/refresh");
+
+        if (!roleService.isSuperAdmin(httpRequest)) {
+            throw new ForbiddenException();
+        }
+
+        User user = jwtService.getUserFromJwt(httpRequest)
+                .orElseThrow(() -> new InvalidRequestException(NO_USER));
+
+        try {
+            spotlightService.refreshToken();
+
+            log.info("Spotlight authorization token successfully refreshed");
+            SpotlightOAuthAudit spotlightOAuthAudit = SpotlightOAuthAudit.builder()
+                    .user(user)
+                    .event(SpotlightOAuthAuditEvent.REFRESH)
+                    .status(SpotlightOAuthAuditStatus.SUCCESS)
+                    .build();
+
+            spotlightService.saveAudit(spotlightOAuthAudit);
+
+            return ResponseEntity.ok().build();
+        } catch (Exception e) {
+            log.error("Error refreshing Spotlight authorization token", e);
+
+            SpotlightOAuthAudit spotlightOAuthAudit = SpotlightOAuthAudit.builder()
+                    .user(user)
+                    .event(SpotlightOAuthAuditEvent.REFRESH)
+                    .status(SpotlightOAuthAuditStatus.FAILURE)
+                    .build();
+
+            spotlightService.saveAudit(spotlightOAuthAudit);
+
+            throw new Exception("Error refreshing Spotlight authorization token");
```

Comment:
I think a dedicated exception might be better here?

Reply:
Fixed, added a dedicated SpotlightTokenException